src/Nethermind/Nethermind.Network/Enode.cs

```diff
@@ -34,9 +35,10 @@ public Enode(PublicKey nodeKey, IPAddress hostIp, int port)
         public Enode(string enodeString)
         {
             string[] enodeParts = enodeString.Split(':');
-            _nodeKey = new PublicKey(enodeParts[1].Split('@')[0].TrimStart('/'));
+            string[] enodeParts2 = enodeParts[1].Split('@');
+            _nodeKey = new PublicKey(enodeParts2[0].TrimStart('/'));
             Port = int.Parse(enodeParts[2]);
-            HostIp = IPAddress.Parse(enodeParts[1].Split('@')[1]);
+            HostIp = IPAddress.TryParse(enodeParts2[1], out var ip) ? ip : Dns.GetHostAddresses(enodeParts2[1]).First();
```

Comment:
seems cool and working (apart from test), I was initially worried that we need to look at other places like NetworkNode and Node but it seems that we just need to get IP address on loading data

Reply:
Yep, we don't need it in other places (for now)